packages/block-editor/src/components/url-input/index.js

```diff
@@ -415,145 +395,82 @@ class URLInput extends Component {
-			value = '',
-			__experimentalShowInitialSuggestions = false,
-		} = this.props;
-
-		const {
-			showSuggestions,
-			suggestions,
-			selectedSuggestion,
-			suggestionsListboxId,
-			suggestionOptionIdPrefix,
-			loading,
-		} = this.state;
-
-		const suggestionsListProps = {
-			id: suggestionsListboxId,
-			ref: this.autocompleteRef,
-			role: 'listbox',
+		return {
+			controlProps,
+			inputProps,
+			isLoading,
 		};
+	}
 
-		const buildSuggestionItemProps = ( suggestion, index ) => {
+	function buildSuggestionsProps() {
+		function buildSuggestionItemProps( _, index ) {
 			return {
 				role: 'option',
 				tabIndex: '-1',
 				id: `${ suggestionOptionIdPrefix }-${ index }`,
-				ref: this.bindSuggestionNode( index ),
+				ref: ( ref ) => {
+					suggestionNodes.current[ index ] = ref;
+				},
 				'aria-selected': index === selectedSuggestion,
 			};
-		};
-
-		if (
-			isFunction( renderSuggestions ) &&
-			showSuggestions &&
-			!! suggestions.length
-		) {
-			return renderSuggestions( {
-				suggestions,
-				selectedSuggestion,
-				suggestionsListProps,
-				buildSuggestionItemProps,
-				isLoading: loading,
-				handleSuggestionClick: this.handleOnClick,
-				isInitialSuggestions:
-					__experimentalShowInitialSuggestions &&
-					! ( value && value.length ),
-			} );
 		}
 
-		if (
-			! isFunction( renderSuggestions ) &&
-			showSuggestions &&
-			!! suggestions.length
-		) {
-			return (
-				<Popover position="bottom" noArrow focusOnMount={ false }>
-					<div
-						{ ...suggestionsListProps }
-						className={ classnames(
-							'block-editor-url-input__suggestions',
-							`${ className }__suggestions`
-						) }
-					>
-						{ suggestions.map( ( suggestion, index ) => (
-							<Button
-								{ ...buildSuggestionItemProps(
-									suggestion,
-									index
-								) }
-								key={ suggestion.id }
-								className={ classnames(
-									'block-editor-url-input__suggestion',
-									{
-										'is-selected':
-											index === selectedSuggestion,
-									}
-								) }
-								onClick={ () =>
-									this.handleOnClick( suggestion )
-								}
-							>
-								{ suggestion.title }
-							</Button>
-						) ) }
-					</div>
-				</Popover>
-			);
-		}
-		return null;
+		return {
+			suggestions,
+			selectedSuggestion,
+			suggestionsListProps: {
+				id: suggestionsListboxId,
+				ref: autocompleteRef,
+				role: 'listbox',
+			},
+			buildSuggestionItemProps,
+			handleSuggestionClick: ( suggestion ) => {
+				selectLink( suggestion );
+				// Move focus to the input field when a link suggestion is clicked.
+				inputRef.current.focus();
+			},
+			isInitialSuggestions: ! value?.length && showInitialSuggestions,
+			isLoading,
+			className,
+		};
 	}
+
+	const controlProps = buildControlProps();
+	const suggestionProps = buildSuggestionsProps();
+
+	return (
+		<>
+			{ renderControl( controlProps ) }
+			{ showSuggestions &&
+				suggestions?.length &&
```

Comment:
`0` can be rendered. https://reactjs.org/docs/jsx-in-depth.html?#booleans-null-and-undefined-are-ignored
```suggestion
				!! suggestions?.length &&
```